docs/redirectless.md

```diff
@@ -0,0 +1,98 @@
+# Redirectless support
+In PingFederate, you can configure mobile applications to authenticate through REST APIs as OAuth clients without needing to handle HTTP redirections. When authentication is complete, the applications receive an OAuth authorization code or access token, and possibly an OpenID Connect ID token.
+
+Single-page web applications can also use redirectless mode if administrators configure them in PingFederate as authentication applications.
+## PingFederate configuration
+1. Follow the instructions at [README.md](../README.md#pingfederate-configuration) to configure Authentication API in PingFederate.
+1. Create a new OAuth 2.0 client at Application > OAuth > Clients
+1. Allow 'allow authentication api oauth initiation' on the newly created OAuth 2.0 client.
+1. Select 'restricted common scopes' required by the client.
+1. Select appropriate grant types.
+
+## Usage
+To make use of the redirctless flow the following steps must be taken:
+- Create an instance of the Authentication widget by providing the base url and the necessray options
+- Create a `configuration` object to provide the necessary redirectless settings. 
+- Call `initRedirectless` and pass the `configuration` object as an argument.
+
+An example is show below: 
+```javascript
+var authnWidget = new PfAuthnWidget("https://localhost", { divId: 'authnwidget' });
+var config = {
+  client_id: 'test',
+  response_type: 'token id_token',
+  onAuthorizationSuccess: function (response) {
+    console.log(response);
+  }
+};
+widget.initRedirectless(config);
+```
+## Configuration object
+There are two ways to create a configuration to initiate the redirectless flow:
+1. The configuration object contians `onAuthorizationSuccess` and the required attributes (e.g. `client_id`, `response_type`, etc.) used by the internal authorization request function.
+1. The configuration object contians `onAuthorizationRequest` and `onAuthorizationSuccess` functions.
+
+Most of deployments should use the first option. The second option is there for advanced use-cases.
+### `onAuthorizationSuccess` function
```

Comment:
These headers for the function names look to be backwards. The info is for the onAuthorizationRequest function